.github/workflows/deploy-to-tp4cf.yml

```diff
@@ -262,7 +262,7 @@ jobs:
 
       - name: Create User Provided Service (if needed by java-spring-ai-mcp/server)
         # This requires the AVIATION_STACK_ACCESS_KEY secret
-        if: inputs.project == 'java-spring-ai-mcp/server' && secrets.AVIATION_STACK_ACCESS_KEY != ''
+        if: inputs.project == 'java-spring-ai-mcp/server' && 'secrets.AVIATION_STACK_ACCESS_KEY' != ''
```

Comment:
All you are doing here is making the secrets reference a literal string by encapsulating it in single-quotes.  I think we need to dig into the overall function of the this Github Action CD job definition. Perhaps refactor into individual jobs that can be delegated to by this master job?